File: PHP/Kode/Televisi.php
<?php
require_once 'ProdukElektronik.php';

// Class turunan dari ProdukElektronik, khusus televisi
class Televisi extends ProdukElektronik {
    // Atribut private
    private $ukuranLayar;
    private $resolusi;
    private $isSmartTV;

    // Konstruktor
    public function __construct($nama = "", $merek = "", $harga = 0, $stok = 0, $foto_produk = "no-image.png", $konsumsiDaya = 0, $voltase = 0, $garansi = 0, $ukuranLayar = 0.0, $resolusi = "", $isSmartTV = false) {
        // Memanggil konstruktor induk
        parent::__construct($nama, $merek, $harga, $stok, $foto_produk, $konsumsiDaya, $voltase, $garansi);
        $this->ukuranLayar = $ukuranLayar;
        $this->resolusi = $resolusi;
        $this->isSmartTV = $isSmartTV;
    }

    // Setter dan Getter untuk setiap atribut
    public function setUkuranLayar($ukuranLayar) {
        $this->ukuranLayar = $ukuranLayar;
    }

    public function getUkuranLayar() {
        return $this->ukuranLayar;
    }

    public function setResolusi($resolusi) {
        $this->resolusi = $resolusi;
    }

    public function getResolusi() {
        return $this->resolusi;
    }

    public function setIsSmartTV($isSmartTV) {
        $this->isSmartTV = $isSmartTV;
    }

    public function getIsSmartTV() {
        return $this->isSmartTV;
    }
}

File: CPP/Kode/Televisi.cpp
#pragma once

#include "ProdukElektronik.cpp"

class Televisi : public ProdukElektronik { // Class turunan dari ProdukElektronik, khusus televisi
    private: // Atribut private
        double ukuranLayar;
        string resolusi;
        bool isSmartTV;

    public: // Method public
        Televisi() { // Konstruktor default
        }

        Televisi(string nama, string merek, int harga, int stok, int konsumsiDaya, int voltase, int garansi,
            double ukuranLayar, string resolusi, bool isSmartTV) 
            : ProdukElektronik(nama, merek, harga, stok, konsumsiDaya, voltase, garansi) { // Memanggil konstruktor induk
            this->ukuranLayar = ukuranLayar;
            this->resolusi = resolusi;
            this->isSmartTV = isSmartTV;
        }

        // Setter dan Getter untuk setiap atribut

        void setUkuranLayar(double ukuranLayar) {
            this->ukuranLayar = ukuranLayar;
        }
        double getUkuranLayar() {
            return this->ukuranLayar;
        }

        void setResolusi(string resolusi) {
            this->resolusi = resolusi;
        }
        string getResolusi() {
            return this->resolusi;
        }

        void setIsSmartTV(bool isSmartTV) {
            this->isSmartTV = isSmartTV;
        }
        bool getIsSmartTV() {
            return this->isSmartTV;
        }

        ~Televisi() { // Destruktor
        }
};